### PR TITLE
bf: allow loops of any size

### PR DESCRIPTION
**Pull request by kazarmy**

### CODE REVIEW

File: libr/anal/p/anal_bf.c [hunk 1]
```diff
@@ -51,12 +53,33 @@ static int bf_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *buf, int len) {
...
-						break;
+						goto beach;
+					}
+				}
+				if (*p == 0x00 || *p == 0xff) {
+					op->type = R_ANAL_OP_TYPE_TRAP;
+					goto beach;
+				}
+				if (i == len - 1 && anal->esil->cb.resize_read_buf) {
+					const ut8 *new_buf;
```
Comment (radare) [inline libr/anal/p/anal_bf.c]: define and assign in the same line

File: libr/anal/p/anal_bf.c [hunk 2]
```diff
@@ -51,12 +53,33 @@ static int bf_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *buf, int len) {
...
+					new_buf = anal->esil->cb.resize_read_buf (anal, len + 1 + BUFSIZE_INC);
+					if (new_buf) {
+						if (buf_resized) {
```
Comment (radare) [inline libr/anal/p/anal_bf.c]: not a big fan of having boolean vars to check if a const* is in reality not const.

File: libr/core/cmd_anal.c [hunk 3]
```diff
@@ -3089,11 +3089,23 @@ static ut64 initializeEsil(RCore *core) {
...
 
+static const ut8 *resize_read_buf(RAnal *anal, int new_len) {
+	RCore *core = anal->user;
+	ut8 *buf = malloc (new_len);
+	if (!buf) {
+		return NULL;
+	}
+	const char *name = r_reg_get_name (anal->reg, R_REG_NAME_PC);
+	ut64 addr = r_reg_getv (anal->reg, name);
+	(void)r_io_read_at (core->io, addr, buf, new_len);
```
Comment (radare) [inline libr/core/cmd_anal.c]: beter use calloc for buf if you dont check return value of read

File: libr/anal/p/anal_bf.c (hunk 2 above)
Comment (kazarmy) [reply libr/anal/p/anal_bf.c]: `buf_resized` is there to differentiate between a buf allocated [on the stack](https://github.com/kazarmy/radare2/blob/6d03ad11f02a7e3b2263b0cb9385386631d35a90/libr/core/cmd_anal.c#L3108) and [on the heap](https://github.com/kazarmy/radare2/blob/6d03ad11f02a7e3b2263b0cb9385386631d35a90/libr/core/cmd_anal.c#L3094) ... doesn't seem to be any other way that is more elegant, unless you want to `strdup (buf)` at start which seems inefficient.

File: libr/anal/p/anal_bf.c (hunk 1 above)
Comment (kazarmy) [reply libr/anal/p/anal_bf.c]: ok

File: libr/core/cmd_anal.c (hunk 3 above)
Comment (kazarmy) [reply libr/core/cmd_anal.c]: done

File: libr/anal/p/anal_bf.c (hunk 2 above)
Comment (kazarmy) [reply libr/anal/p/anal_bf.c]: nvm, removed `buf_resized` in 7881afc5c26afde4b465250568c40c21bd1ea470

**Pull request merged**